coalib/misc/Exceptions.py

```diff
@@ -15,6 +17,12 @@ def get_exitcode(exception, log_printer=None):
         exitcode = 0
     elif isinstance(exception, SystemExit):
         exitcode = exception.code
+    elif (isinstance(exception, ContextualVersionConflict) or
+          isinstance(exception, VersionConflict)):
+        log_message = Constants.VERSION_CONFLICT_MESSAGE
+        log_message += exception.report()
+        log_printer.log_exception(log_message, exception)
+        exitcode = 4
```

Comment:
@sils1297 ? Could you take a call on this?

-----Original Message-----
From: "Adhityaa Chandrasekar" <notifications@github.com>
Sent: ‎17-‎03-‎2016 17:33
To: "coala-analyzer/coala" <coala@noreply.github.com>
Cc: "AbdealiJK" <abdealikothari@gmail.com>
Subject: Re: [coala] Exceptions: Handle package version conflicts (#1894)

In coalib/misc/Exceptions.py:
> @@ -15,6 +17,12 @@ def get_exitcode(exception, log_printer=None):
>          exitcode = 0
>      elif isinstance(exception, SystemExit):
>          exitcode = exception.code
> +    elif (isinstance(exception, ContextualVersionConflict) or
> +          isinstance(exception, VersionConflict)):
> +        log_message = Constants.VERSION_CONFLICT_MESSAGE
> +        log_message += exception.report()
> +        log_printer.log_exception(log_message, exception)
> +        exitcode = 4
It's not really new: Docs - Just felt that it's another version related problem. But I think this deserves its own exit code. How about 13? It's used by Java.
—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub